pkg/arvo/app/observe-hook.hoon

```diff
@@ -260,9 +288,53 @@
     ==  == 
   --
 ::
+++  on-arvo
+  |=  [=wire =sign-arvo]
+  ^-  (quip card _this)
+  ?+    wire  (on-arvo:def wire sign-arvo)
+      [%warm-cache ~]
+    ?.  warm-cache
+      [~ this]
+    ?>  ?=([%clay %writ *] sign-arvo)
+    :_  this
+    :-  =/  =rave:clay  [%next [%t da+now.bowl /mar]]
+        [%pass /warm-cache %arvo %c %warp our.bowl %home `rave]
+    =*  riot  p.sign-arvo
+    ?~  riot
+      ~
+    %+  turn  !<((list path) q.r.u.riot)
+    |=  pax=path
+    ^-  card
+    =.  pax  `path`(snip (slag 1 `(list @)`pax))
+    =/  mark=@ta
+      %+  roll  pax
+      |=  [=term mark=term]
+      ?:  =(0 (lent (trip mark)))
+        term
+      %-  crip
+      %-  zing
+      :~  (trip mark)
+          "-"
+          (trip term)
+      ==
+    =/  =rave:clay  [%sing %b [%da now.bowl] [mark ~]]
+    [%pass [%mar mark ~] %arvo %c %warp our.bowl %home `rave]
+  ::
+      [%mar ^]
+    ?.  warm-cache
+      [~ this]
+    ?>  ?=([%clay %writ *] sign-arvo)
+    =*  riot  p.sign-arvo
+    =*  mark  t.wire
+    ?~  riot
+      ~|('mark build failed for {<mark>}' !!)
```

Comment:
I'm a little surprised this doesn't crash when you delete a mark.  In that case I would expect a `%next` subscription to give a null result.

Reply:
It does crash upon a mark being deleted, but afterward doesn't rebuild the mark or continue to listen to that mark. Would you prefer we just return ~?

Reply:
Yeah probably.  We shouldn't print stack traces when it's probably not an error.

Reply:
Also you probably wanted double quotes, though if the message is going away the point is moot.